experimental/segmenter/src/lstm_bies.rs

```diff
@@ -140,23 +146,24 @@ impl<'l> Lstm<'l> {
         // in the embedding layer of the model.
         // Already checked that the name of the model is either "codepoints" or "graphclsut"
         // TODO: Avoid allocating a string for each code point
-        let input_seq: Vec<i16> = if self.data.get().model.contains("_codepoints_") {
+        let input_seq: Vec<i16> = if let Some(grapheme) = self.grapheme {
+            grapheme
+                .segment_str(input)
+                .collect::<Vec<usize>>()
+                .windows(2)
+                .map(|chunk| {
+                    self.return_id(
+                        input
+                            .get(*chunk.get(0).unwrap_or(&0)..*chunk.get(1).unwrap_or(&input.len()))
+                            .unwrap_or(input),
+                    )
+                })
+                .collect()
```

Comment:
`optional`: This feels a bit crude.

Can you use something like:

```rust
let low = graphme.segment_str(input);
let high = low.clone();
let input_seq = low.zip(high.skip(1)).map(|(a, b)| {
    self.return_id(input.get(a..b).unwrap_or(input))
}).collect();
```

Playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b3e128908ff62916b46cf4fa47942aeb